server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServerResourceManager.java

```diff
@@ -449,7 +584,33 @@ private void manageMemory() {
           mma = memoryManager.getMemoryManagementActions(tabletStates);
 
         } catch (Throwable t) {
+
           log.error("Memory manager failed {}", t.getMessage(), t);
+
+          // treat error as unrecoverable - get zookeeper and delete server lock to kill instance.
```

Comment:
If a tablet server needs to kill itself, it would probably be better to halt rather than deleting the lock.  If things are not going well inside the process, deleting the lock may not kill the process.  Halting probably has a better chance.